playbooks/prerun-network-checks.yml

```diff
@@ -1,188 +1,199 @@
+# the default('localhost') is only useful while testing on molecule
 - name: Run preliminary network checks for repository hosts
   hosts: repository
   become: true
-  roles:
-    - role: '../roles/helper_modules'
   tasks:
+    - name: Include common defaults
+      ansible.builtin.include_vars: ../roles/common/defaults/main.yml
+
     - name: Check db connection
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.database | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.database[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+        checked_host: "{{ hostvars[groups.database[0]].ansible_host | default('localhost') }}"
         checked_port: "{{ ports_cfg.postgres.sql }}"
         delegate_target: "{{ groups.database | first }}"
-      when: repo_db_url == ""
+      when: repo_db_url == "" and groups.database | default([]) | length > 0
 
     - name: Check activemq connection
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.activemq | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.activemq[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+        checked_host: "{{ hostvars[groups.activemq[0]].ansible_host | default('localhost') }}"
         checked_port: "{{ ports_cfg.activemq[activemq_protocol] }}"
         delegate_target: "{{ groups.activemq | first }}"
       when: groups.activemq | default([]) | length > 0
 
     - name: Check search connection
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.search | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.search[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+        checked_host: "{{ hostvars[groups.search[0]].ansible_host | default('localhost') }}"
         checked_port: "{{ ports_cfg.search.http }}"
         delegate_target: "{{ groups.search | first }}"
       when: groups.search | default([]) | length > 0
 
     - name: Check sync connection
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.syncservice | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.syncservice[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+        checked_host: "{{ hostvars[groups.syncservice[0]].ansible_host | default('localhost') }}"
         checked_port: "{{ ports_cfg.sync.http }}"
         delegate_target: "{{ groups.syncservice | first }}"
       when:
         - groups.syncservice | default([]) | length > 0
-        - acs.edition == "Enterprise"
```

Comment:
Why are you excluding the enterprise ersion condition for sync and adding it for ats?